task-1/src/Point.ts

```diff
@@ -0,0 +1,34 @@
+export class Point {
+    private x: number;
+    private y: number;
+
+    constructor();
+    constructor(x: number, y: number);
+    constructor(x?: number, y?: number) {
+        this.x = x ?? 0;
+        this.y = y ?? 0;
+    }
+
+    toString(): string {
+        return `(${this.x}, ${this.y})`;
+    }
+
+    private getDistance(point1: Point, point2: Point): number {
```

Comment:
I think it's better to make this method static because it's not for working with a particular instance, rather than with any Point in general. Especially, `this.getDistance(this, point)` looks pretty weird😅